File: src/main/java/com/almasb/zeph/combat/Element.java
package com.almasb.zeph.combat;

public enum Element {
    NEUTRAL(1.00f, 0.75f, 0.75f, 0.75f, 0.75f),
    FIRE   (1.25f, 0.00f, 0.25f, 0.50f, 2.00f),
    WATER  (1.25f, 2.00f, 0.00f, 0.25f, 0.50f),
    AIR    (1.25f, 0.50f, 2.00f, 0.00f, 0.25f),
    EARTH  (1.25f, 0.25f, 0.50f, 2.00f, 0.00f);

    private float[] modifiers;

    public float getDamageModifierAgainst(Element element) {
        return this.modifiers[element.ordinal()];
    }

    Element(float... modifiers) {
        this.modifiers = modifiers;
    }
}

File: src/main/kotlin/com/almasb/zeph/combat/Element.kt
package com.almasb.zeph.combat

import javafx.scene.paint.Color

/**
 * There are 5 element types.
 *
 * @author Almas Baimagambetov (almaslvl@gmail.com)
 */
enum class Element(vararg modifiers: Double) {
    NEUTRAL(1.00, 0.75, 0.75, 0.75, 0.75),
    FIRE   (1.25, 0.00, 0.25, 0.50, 2.00),
    WATER  (1.25, 2.00, 0.00, 0.25, 0.50),
    AIR    (1.25, 0.50, 2.00, 0.00, 0.25),
    EARTH  (1.25, 0.25, 0.50, 2.00, 0.00);

    private val modifiers: DoubleArray = modifiers

    val color: Color
        get() = when (this) {
            NEUTRAL -> Color.DARKGRAY
            FIRE -> Color.RED
            WATER -> Color.BLUE
            AIR -> Color.AZURE
            EARTH -> Color.BROWN
        }

    fun getDamageModifierAgainst(element: Element): Double {
        return this.modifiers[element.ordinal]
    }
}